### PR TITLE
staging_feat(Navigation): Handle nav item with no associated page

### LINKED ISSUE #207: Handle edge case where nav item has no associated page or external link

### PR DESCRIPTION
**Pull request by m7kvqbe1**
## Related issue

Closes #207

## Overview

Handle edge case where a navigation item has no associated page.

## Reason

>Navigation and pages do not always have a like-for-like relationship.

## Work carried out

- [x] Refactor `getStaticPaths` and `getStaticProps` data fetching
- [x] Handle Nav item with no page edge case
- [x] Added e2e Cypress test

### CODE REVIEW

File: pages/[...slug].tsx [hunk 1]
```diff
@@ -91,27 +79,15 @@ async function fetchPageByPath(path: string): Promise<NavigationElementType> {
...
-  const path = Array.isArray(slug) ? `/${slug.join('/')}` : `/${slug}`
-
-  const { page } = await fetchPageByPath(path)
-
-  const {
-    childrenCollection: { items },
-  } = await fetchPageByPath(`/${slug[0]}`)
+async function fetchDesktopNavigation(): Promise<NavigationType> {
+  const { navigation } = await contentful(NAVIGATION_BY_ID_QUERY, {
+    id: '3dJ4ZZB9rMxXS4oe2iKuEY',
```
Comment (thyhjwb6) [inline pages/[...slug].tsx:87]: Making this ID more self documenting would be useful.

File: pages/[...slug].tsx [hunk 2]
```diff
@@ -291,27 +282,38 @@ export const GenericPage: React.FC<GenericPageProps> = ({
...
+          }) => {
+            const parentHref = parentPage
+              ? parentPath || '#'
+              : parentExternalUri || '#'
+
             return (
               <MastheadMenuItem
                 key={parentPath}
-                link={<Link href={parentPath}>{parentTitle}</Link>}
+                link={<Link href={parentHref}>{parentTitle}</Link>}
```
Comment (thyhjwb6) [inline pages/[...slug].tsx:299]: I wonder if this could be `href={parentHref || '#'}` which reduces the complexity for [setting `parentHref`](https://github.com/Royal-Navy/docs.royalnavy.io/pull/218/files#diff-5e5186614071390e5a8824f814c2a49cd4083adc99ada4220e546a421676ca2aR292-R294).

File: pages/[...slug].tsx [hunk 3]
```diff
@@ -291,27 +282,38 @@ export const GenericPage: React.FC<GenericPageProps> = ({
...
+
                       return (
                         <MastheadSubMenuItem
                           key={childPath}
-                          link={
-                            <Link href={childPath || childExternalUri}>
-                              {childTitle}
-                            </Link>
-                          }
+                          link={<Link href={childHref}>{childTitle}</Link>}
```
Comment (thyhjwb6) [inline pages/[...slug].tsx:316]: Same suggestion here for reducing complexity.

**Review by thyhjwb6 — APPROVED** [review]
Looks good, I have approved. A couple of suggestions you can choose to use.
